README.md

```diff
@@ -130,83 +134,98 @@ const ipld = new Ipld({
 })
 ```
 
-##### `options.loadFormat(codec, callback)`
+##### `options.loadFormat(codec)`
 
 | Type | Default |
 |------|---------|
-| `Function` | `null` |
+| `async Function` | `null` |
 
-Function to dynamically load an [IPLD Format](https://github.com/ipld/interface-ipld-format). It is passed a string `codec`, the multicodec of the IPLD format to load and a callback function to call when the format has been loaded. e.g.
+Function to dynamically load an [IPLD Format](https://github.com/ipld/interface-ipld-format). It is passed a `codec`, the multicodec code of the IPLD format to load and returns an IPLD Format implementation. For example:
 
 ```js
+const multicodec = require('multicodec')
 const ipld = new Ipld({
-  loadFormat (codec, callback) {
-    if (codec === 'git-raw') {
-      callback(null, require('ipld-git'))
+  async loadFormat (codec) {
+    if (codec === multicodec.GIT_RAW) {
+      return require('ipld-git')
     } else {
-      callback(new Error('unable to load format ' + codec))
+      throw new Error('unable to load format ' + multicodec.print[codec])
     }
   }
 })
 ```
 
-### `.put(node, options, callback)`
+### `.put(nodes, options)`
+
+> Stores the given IPLD Nodes of a recognized IPLD Format.
+
+ - `nodes` (`Iterable<Object>`): deserialized IPLD nodes that should be inserted.
+
+ - `options` is applied to any of the `nodes` and is an object with the following properties:
+   - `format` (`multicodec`, required): the multicodec of the format that IPLD Node should be encoded in.
+   - `hashAlg` (`multicodec`, default: hash algorithm of the given multicodec): the hashing algorithm that is used to calculate the CID.
+   - `version` (`number`, default: 1): the CID version to use.
+   - `onlyHash` (`boolean`, default: false): if true the serialized form of the IPLD Node will not be passed to the underlying block store.
+
+Returns an async iterator with the CIDs of the serialized IPLD Nodes.
+
+
+### `.resolve(cid, path)`
+
+> Retrieves IPLD Nodes along the `path` that is rooted at `cid`.
 
-> Store the given node of a recognized IPLD Format.
+ - `cid` (`CID`, required): the CID the resolving starts.
+ - `path` (`IPLD Path`, required): the path that should be resolved.
```

Comment:
This is the point. `resolve()` and `get()` return different things. `get()` returns whatever the IPLD Format is creating out of it. `resolve()` returns the data as if you call `tree()` on it and assemble the data. Though hopefully, those two things might be the same once the new IPLD Format API is in, but until then I'd like to keep the distinction.

Reply:
Would it allow empty string?

Reply:
> Would it allow empty string?

That was my plan. Though it would only work with the new IPLD Format API, it would be too much hassle to get it working with the old formats.

Though I envisioned that people would use `resolve()` to get a single value or a subset of something (or deeply nested) and would use `get()` if they want the full object.